vscode/src/non-stop/FixupController.ts

```diff
@@ -305,6 +309,15 @@ export class FixupController
             const codeCount = countCode(replacementText)
             const source = task.source
             telemetryService.log('CodyVSCodeExtension:fixup:applied', { ...codeCount, source })
+
+            // This is currently a test/demo of the new events system
+            telemetryRecorder.recordEvent('cody.fixup.apply', 'succeeded', {
+                metadata: [
+                    ['lineCount', codeCount.lineCount],
+                    ['charCount', codeCount.charCount],
+                ],
```

Comment:
```suggestion
                metadata: {
                    lineCount: codeCount.lineCount,
                    charCount: codeCount.charCount,
                },
```

Unrelated to the above question: wouldn't this be nicer as an object (won't allow duplicated definitions for the same argument, for example)

Reply:
It's not possible, I think, to require object keys to be from a particular type or enum

Reply:
In particular, when I tried to explore an option like this I found that Typescript does not check against excess properties, so callers can assign whatever they want into keys

Reply:
This seems to work 🙈  Maybe I miss something though? https://www.typescriptlang.org/play?#code/KYOwrgtgBAYg9nKBvAUFWCCMUC8UBE8cm+ANGhnAEy4FFVkoC+KKAZmCAMYAuAlnBBQANnADmACjgAjAFYAuZFADaAa2ABPKHyFEoAHygByaQEMATkYC6i8BGnBzUJgEokLFKMmp06ZUQA6IkwbKExSKAo-QPpQqnJfKDNzRQBmBN9OVRA4AHcQRQAWZhcgA

Reply:
Wow, this is why I need the pros in the room 😁 thank you, let me play around with this!

Reply:
This works great, can't believe I missed it 😆 Updated in https://github.com/sourcegraph/telemetry/commit/78682201b30fa9fe0ae30394cf376d0ec32a2e3f and in this PR as well, thank you!